tailcfg/tailcfg.go

```diff
@@ -192,19 +190,66 @@ type Node struct {
 	MachineAuthorized bool `json:",omitempty"` // TODO(crawshaw): replace with MachineStatus
 }
 
-// DefaultDisplayName returns a value suitable
-// for using as the default value for n.DisplayName.
-func (n *Node) DefaultDisplayName() string {
-	if n.Name != "" {
-		// Use the Magic DNS prefix as the default display name.
-		return dnsname.ToBaseName(n.Name)
+// displayName holds the various names that can be used for this node in UIs.
+type displayName struct {
+	Name     string // primary name for the node
+	Subname  string // subname for the node; may be empty
+	FullName string // in format "name (subname)" when subname is non-empty, otherwise just "name"
+}
+
+// DisplayName returns the user-facing name for a
```

Comment:
remember to document on these that they're only valid if InitDisplayNames has been called.